drivers/dma/dma_stm32u5.c

```diff
@@ -0,0 +1,757 @@
+void dma_stm32_clear_te(DMA_TypeDef *DMAx, uint32_t id)
+{
+	LL_DMA_ClearFlag_DTE(DMAx, dma_stm32_id_to_stream(id));
+	LL_DMA_ClearFlag_ULE(DMAx, dma_stm32_id_to_stream(id));
+	LL_DMA_ClearFlag_USE(DMAx, dma_stm32_id_to_stream(id));
+}
+
+bool dma_stm32_is_ht_active(DMA_TypeDef *DMAx, uint32_t id)
+{
+	return LL_DMA_IsActiveFlag_HT(DMAx, dma_stm32_id_to_stream(id));
+}
+
+void dma_stm32_clear_ht(DMA_TypeDef *DMAx, uint32_t id)
+{
+	LL_DMA_ClearFlag_HT(DMAx, dma_stm32_id_to_stream(id));
+}
+
+void stm32_dma_dump_stream_irq(DMA_TypeDef *dma, uint32_t id)
+{
+	LOG_INF("tc: %d, ht: %d, te: %d",
+		dma_stm32_is_tc_active(dma, id),
+		dma_stm32_is_ht_active(dma, id),
+		dma_stm32_is_te_active(dma, id));
+}
+
+/* Check if nsecure masked interrupt is active on channel */
+bool stm32_dma_is_tc_irq_active(DMA_TypeDef *dma, uint32_t id)
+{
+	return (LL_DMA_IsEnabledIT_TC(dma, dma_stm32_id_to_stream(id)) &&
+		LL_DMA_IsActiveFlag_TC(dma, dma_stm32_id_to_stream(id)));
+}
+
+bool stm32_dma_is_ht_irq_active(DMA_TypeDef *dma, uint32_t id)
+{
+	return (LL_DMA_IsEnabledIT_HT(dma, dma_stm32_id_to_stream(id)) &&
+		LL_DMA_IsActiveFlag_HT(dma, dma_stm32_id_to_stream(id)));
+}
+
+static inline bool stm32_dma_is_te_irq_active(DMA_TypeDef *dma, uint32_t id)
+{
+	return (
+		(LL_DMA_IsEnabledIT_DTE(dma, dma_stm32_id_to_stream(id)) &&
+		LL_DMA_IsActiveFlag_DTE(dma, dma_stm32_id_to_stream(id))) ||
+		(LL_DMA_IsEnabledIT_ULE(dma, dma_stm32_id_to_stream(id)) &&
+		LL_DMA_IsActiveFlag_ULE(dma, dma_stm32_id_to_stream(id))) ||
+		(LL_DMA_IsEnabledIT_USE(dma, dma_stm32_id_to_stream(id)) &&
+		LL_DMA_IsActiveFlag_USE(dma, dma_stm32_id_to_stream(id)))
+		);
+}
+
+/* check if and irq of any type occurred on the channel */
+#define stm32_dma_is_irq_active LL_DMA_IsActiveFlag_MIS
+
+void stm32_dma_clear_stream_irq(DMA_TypeDef *dma, uint32_t id)
+{
+	dma_stm32_clear_te(dma, id);
+
+	LL_DMA_ClearFlag_TO(dma, dma_stm32_id_to_stream(id));
+	LL_DMA_ClearFlag_SUSP(dma, dma_stm32_id_to_stream(id));
+}
+
+bool stm32_dma_is_irq_happened(DMA_TypeDef *dma, uint32_t id)
+{
+	if (dma_stm32_is_te_active(dma, id)) {
+		return true;
+	}
+
+	return false;
+}
+
+bool stm32_dma_is_unexpected_irq_happened(DMA_TypeDef *dma, uint32_t id)
+{
+	/* Preserve for future amending. */
+	return false;
+}
+
+void stm32_dma_enable_stream(DMA_TypeDef *dma, uint32_t id)
+{
+	LL_DMA_EnableChannel(dma, dma_stm32_id_to_stream(id));
+}
+
+int stm32_dma_disable_stream(DMA_TypeDef *dma, uint32_t id)
+{
+	/* GPDMA channel abort sequence */
+	LL_DMA_SuspendChannel(dma, dma_stm32_id_to_stream(id));
+
+	/* reset the channel will disable it */
+	LL_DMA_ResetChannel(dma, dma_stm32_id_to_stream(id));
+
+	if (!LL_DMA_IsEnabledChannel(dma, dma_stm32_id_to_stream(id))) {
+		return 0;
+	}
+
+	return -EAGAIN;
+}
+
+void stm32_dma_set_mem_periph_address(DMA_TypeDef *dma,
+					     uint32_t channel,
+					     uint32_t src_addr,
+					     uint32_t dest_addr)
+{
+	LL_DMA_ConfigAddresses(dma, channel, src_addr, dest_addr);
+}
```

Comment:
I get this is inherited from others drivers, but in case of this specific driver that it mean to be U5 specific, this level abstraction sounds useless

Reply:
ok for code simplification